xchange-latoken/src/main/java/org/knowm/xchange/latoken/service/LatokenTradeService.java

```diff
@@ -0,0 +1,249 @@
+package org.knowm.xchange.latoken.service;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.knowm.xchange.Exchange;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.Order.IOrderFlags;
+import org.knowm.xchange.dto.trade.LimitOrder;
+import org.knowm.xchange.dto.trade.OpenOrders;
+import org.knowm.xchange.dto.trade.UserTrades;
+import org.knowm.xchange.exceptions.CurrencyPairNotValidException;
+import org.knowm.xchange.exceptions.ExchangeException;
+import org.knowm.xchange.latoken.LatokenAdapters;
+import org.knowm.xchange.latoken.LatokenErrorAdapter;
+import org.knowm.xchange.latoken.dto.LatokenException;
+import org.knowm.xchange.latoken.dto.trade.LatokenCancelledOrders;
+import org.knowm.xchange.latoken.dto.trade.LatokenNewOrder;
+import org.knowm.xchange.latoken.dto.trade.LatokenOrder;
+import org.knowm.xchange.latoken.dto.trade.LatokenOrderStatus;
+import org.knowm.xchange.latoken.dto.trade.LatokenTestOrder;
+import org.knowm.xchange.latoken.dto.trade.LatokenUserTrades;
+import org.knowm.xchange.service.trade.TradeService;
+import org.knowm.xchange.service.trade.params.CancelOrderByIdParams;
+import org.knowm.xchange.service.trade.params.CancelOrderParams;
+import org.knowm.xchange.service.trade.params.TradeHistoryParamCurrencyPair;
+import org.knowm.xchange.service.trade.params.TradeHistoryParamLimit;
+import org.knowm.xchange.service.trade.params.TradeHistoryParams;
+import org.knowm.xchange.service.trade.params.orders.DefaultOpenOrdersParamCurrencyPair;
+import org.knowm.xchange.service.trade.params.orders.OpenOrdersParamCurrencyPair;
+import org.knowm.xchange.service.trade.params.orders.OpenOrdersParams;
+import org.knowm.xchange.service.trade.params.orders.OrderQueryParamCurrencyPair;
+import org.knowm.xchange.service.trade.params.orders.OrderQueryParams;
+
+public class LatokenTradeService extends LatokenTradeServiceRaw implements TradeService {
+
+	public LatokenTradeService(Exchange exchange) {
+
+		super(exchange);
+	}
+
+	@Override
+    public OpenOrders getOpenOrders(OpenOrdersParams params) throws IOException {
+		
+    	try {
+    		if (params instanceof OpenOrdersParamCurrencyPair == false) {
+    			throw new CurrencyPairNotValidException("CurrencyPair is must be provided to get open orders.");
```

Comment:
```
/**
 * Exception indicating that a request was made with a {@link CurrencyPair} that is not supported on
 * this exchange.
 *
 * @author bryant_harris
 */
public class CurrencyPairNotValidException extends ExchangeException {
```

not providing it in a method that requires it is a different think, just use `ExchangeException` here and below